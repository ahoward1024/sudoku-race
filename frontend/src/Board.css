:root {
  --color-light-border: #888888;
  --color-dark-border: #455496;
}

.table-style {
  border-collapse: collapse;
  border-spacing: 0;
  position: fixed;
  width: 80vh;
  height: 80vh;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.cell {
  border-top: var(--color-light-border) solid 1px;
  border-bottom: var(--color-light-border) solid 1px;
  border-left: var(--color-light-border) solid 1px;
  border-right:  var(--color-light-border) solid 1px;
  vertical-align: middle;
  text-align: center;
  margin: 0;
  padding: 0pt;
  border-spacing: 0;
  border-collapse: collapse;
  width: 50pt;
  height: 50pt;
}

.cell.tl {
  border-top: var(--color-dark-border) solid 2px;
  border-left: var(--color-dark-border) solid 2px;
}

.cell.tm {
  border-top: var(--color-dark-border) solid 2px;
}

.cell.tr {
  border-top: var(--color-dark-border) solid 2px;
  border-right: var(--color-dark-border) solid 2px;
}

.cell.ml {
  border-left: var(--color-dark-border) solid 2px;
}

.cell.mm {

}

.cell.mr {
  border-right: var(--color-dark-border) solid 2px;
}

.cell.bl {
  border-bottom: var(--color-dark-border) solid 2px;
  border-left: var(--color-dark-border) solid 2px;
}

.cell.bm {
  border-bottom: var(--color-dark-border) solid 2px;
}

.cell.br {
  border-bottom: var(--color-dark-border) solid 2px;
  border-right: var(--color-dark-border) solid 2px;
}


.cell-input {
  border-collapse: collapse;
  border: solid 0 #fff;
  outline: none;
  text-align: center;
  width: 90%;
  height: 90%;
  font-size: 200%;
}
