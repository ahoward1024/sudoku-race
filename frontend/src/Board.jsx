import React, {Component} from 'react';
import './Board.css';
import InputCell from './InputCell';
import NoInputCell from './NoInputCell';

class Board extends Component {
  constructor(props) {
    super(props);
    this.state = {
      'board': '',
      'gameid': ''
    };
    this.getCellValues = this.getCellValues.bind(this);
  }

  componentDidMount() {
    fetch('http://backend.sudokurace.io/game.create')
      .then(resp => resp.json())
      .then(json => json.board.split(''))
      .then(board => this.setState({board}))
      .catch(resp => {
        console.log(`Error getting response from server: ${resp}`);
      });
  }

  getCellValues() {
    const rows = [];
    let index = 0,
        it = 1;
    for (let perRow = 0; perRow < 9; perRow += 1) {
      const row = [];
      for (let perColumn = 0; perColumn < 9; perColumn += 1) {
        const value = this.state.board[index];
        // NOTE/FIXME(alex): Without (val === undefined) React will complain that the CellNoInput
        // is an uncontrolled component that is being switched to a controlled component.
        if (value === ' ' || value === undefined) {
          row.push(<td key={`c${perRow}${perColumn}`} className={`cell c${it}`}>
                   <InputCell value={`${value}`} index={`${index}`}/></td>);
        } else {
          row.push(<td key={`c${perRow}${perColumn}`} className={`cell c${it}`}>
                   <NoInputCell value={`${value}`} index={`${index}`}/></td>);
        }
        if (it % 3 === 0) {
          it -= 3;
        }
        it += 1;
        index += 1;
      }
      it += 3;
      if (it % 10 === 0) {
        it -= 9;
      }
      rows.push(<tr key={`r${perRow}`}>{row}</tr>);
    }

    return rows;
  }

  render() {
    const rows = this.getCellValues();

    return (
      <div >
        <table className="table-style">
          <tbody className="cell">{rows}</tbody>
        </table>
      </div>
    );
  }
}

export default Board;
