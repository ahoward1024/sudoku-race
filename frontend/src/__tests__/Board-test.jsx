import React from 'react';
import {shallow} from 'enzyme';
import Board from '../Board';
import renderer from 'react-test-renderer';

test('Test the Board component', () => {
  const component = renderer.create(
    <Board width="400" height="400"/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
