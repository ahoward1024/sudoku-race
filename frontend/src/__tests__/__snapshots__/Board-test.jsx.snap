// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Board component 1`] = `
<div>
  <table
    className="table-style"
    style={
      Object {
        "height": "400",
        "width": "400",
      }
    }
  >
    <tbody
      className="cell"
    >
      <tr>
        <td
          className="cell c1"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c2"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c3"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c1"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c2"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c3"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c1"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c2"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c3"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
      </tr>
      <tr>
        <td
          className="cell c4"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c5"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c6"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c4"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c5"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c6"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c4"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c5"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c6"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
      </tr>
      <tr>
        <td
          className="cell c7"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c8"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c9"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c7"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c8"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c9"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c7"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c8"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c9"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
      </tr>
      <tr>
        <td
          className="cell c1"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c2"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c3"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c1"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c2"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c3"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c1"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c2"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c3"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
      </tr>
      <tr>
        <td
          className="cell c4"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c5"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c6"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c4"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c5"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c6"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c4"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c5"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c6"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
      </tr>
      <tr>
        <td
          className="cell c7"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c8"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c9"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c7"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c8"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c9"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c7"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c8"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c9"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
      </tr>
      <tr>
        <td
          className="cell c1"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c2"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c3"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c1"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c2"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c3"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c1"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c2"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c3"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
      </tr>
      <tr>
        <td
          className="cell c4"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c5"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c6"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c4"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c5"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c6"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c4"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c5"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c6"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
      </tr>
      <tr>
        <td
          className="cell c7"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c8"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c9"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c7"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c8"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c9"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c7"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c8"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
        <td
          className="cell c9"
        >
          <input
            className="cell-input"
            defaultValue=""
            maxLength="1"
            onKeyPress={[Function]}
            type="text"
          />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
